### DISCUSSION TITLE
Events all node question?

### DISCUSSION THREAD

**Discussion by stevegroom** [zachowj/node-red-contrib-home-assistant-websocket#1246] [General]
I’m using node red for most of my HA automations. Each flow seems to run fast enough ( HA Yellow) but some flows are a bit slow to get started.  

I was thinking about the way the event: all node works - from my understanding the node is triggered for every event or at least every event relating to an event class - I’m focusing on ZHA_event - the event: all nodes then have to filter to select device_id or device_ieee before continuing.

Does this mean that all nodes listening on ZHA_event is dispatched for each event and the most of them check and see they have no work to perform? In my set up, there are currently 42 Events: all nodes. 

Would having a single events:all node and then a switch node to dispatch to the appropriate flow perform significantly better than having many nodes? 

Steve

**Comment by zachowj** [comment]
Your understanding is correct for the functionality of the events: all nodes. I haven't run any performant tests but it should only be taking milliseconds for each event to be checked if it's valid. I would be surprised if you saw an impact even with 1000 events: all nodes.

If you were listening for the same ZHA event and device over several events: all, would it be faster to have a single event: all node to do checking then use a link node or something similar to push the events to different flows? Sure, but I don't think it would be perceptual to a human.
